id-repository/id-repository-identity-service/src/main/java/io/mosip/idrepository/identity/helper/IdRepoServiceHelper.java

```diff
@@ -146,29 +143,43 @@ public String getSchema(String schemaVersion) {
-                        requestMap.get(selectedHandlesFieldId));
-                List<String> selectedHandleFieldIds = (List<String>) identityMap.get(selectedHandlesFieldId);
-                return selectedHandleFieldIds.stream()
-                        .filter( handleFieldId -> supportedHandlesInSchema.get(schemaVersion).contains(handleFieldId))
-                        .collect(Collectors.toMap(handleName->handleName,
-                                handleFieldId-> {
-                                    String handle = ((String) identityMap.get(handleFieldId))
-                                            .concat(getHandlePostfix(handleFieldId))
-                                            .toLowerCase(Locale.ROOT);
-                                    return new HandleDto(handle, getHandleHash(handle));
-                                }));
-            }
-        }
-        return Map.of();
-    }
+	/**
+	 * @param identity it should contain the data for attributes coming as part of
+	 *                 selected handles.
+	 * @return handles map
+	 * @throws IdRepoAppException
+	 */
+	public Map<String, HandleDto> getSelectedHandles(Object identity) throws IdRepoAppException {
+		Map<String, Object> requestMap = convertToMap(identity);
+		if (requestMap.containsKey(ROOT_PATH) && Objects.nonNull(requestMap.get(ROOT_PATH))) {
+			Map<String, Object> identityMap = (Map<String, Object>) requestMap.get(ROOT_PATH);
+			String schemaVersion = String
+					.valueOf(identityMap.get(identityMapping.getIdentity().getIDSchemaVersion().getValue()));
+			String selectedHandlesFieldId = identityMapping.getIdentity().getSelectedHandles().getValue();
+			if (identityMap.containsKey(selectedHandlesFieldId)
+					&& Objects.nonNull(identityMap.get(selectedHandlesFieldId))) {
+				mosipLogger.debug(IdRepoSecurityManager.getUser(), ID_REPO_SERVICE_HELPER, "getSelectedHandles",
+						identityMap.get(selectedHandlesFieldId));
+				List<String> selectedHandleFieldIds = (List<String>) identityMap.get(selectedHandlesFieldId);
+				return selectedHandleFieldIds.stream()
+						.filter(handleFieldId -> supportedHandlesInSchema.get(schemaVersion).contains(handleFieldId))
+						.collect(Collectors.toMap(handleName -> handleName, handleFieldId -> {
+							try {
+								String handle = ((String) identityMap.get(handleFieldId))
+										.concat(getHandlePostfix(handleFieldId)).toLowerCase(Locale.ROOT);
+								return new HandleDto(handle, getHandleHash(handle));
+							} catch (NullPointerException e) {
+								mosipLogger.error(IdRepoSecurityManager.getUser(), ID_REPO_SERVICE_HELPER,
+										"getSelectedHandles", "\n" + handleFieldId + " is null");
+								throw new IdRepoAppUncheckedException(MISSING_INPUT_PARAMETER.getErrorCode(),
+										String.format(MISSING_INPUT_PARAMETER.getErrorMessage(),
+												ROOT_PATH + IdObjectValidatorConstant.PATH_SEPERATOR + handleFieldId));
+							}
+						}));
+			}
+		}
+		return null;
```

Comment:
Instead of null, it recommended returning an empty map, so that we don't have to worry about null check in the caller method.

Reply:
here I am returning 'null' for scenarios where 'selectedHandles' is not coming in the request body itself that means there will be no change in handles. And 'empty map' is for those scenarios where 'selectedHandles' will come as empty list in request body that means we need to revoke all existing handles.